src/query/expression/src/kernels/take.rs

```diff
@@ -116,6 +117,19 @@ where I: databend_common_column::types::Index
         Ok(())
     }
 
+    fn visit_column(&mut self, column: Column) -> Result<()> {
+        match column {
+            Column::Date(buffer) => self.visit_simple_type::<DateType>(buffer),
+            Column::Timestamp(buffer) => self.visit_simple_type::<TimestampType>(buffer),
+            Column::Number(number) => {
+                with_number_mapped_type!(|NUM_TYPE| match number {
+                    NumberColumn::NUM_TYPE(b) => self.visit_simple_type::<NumberType<NUM_TYPE>>(b),
+                })
+            }
+            _ => Self::default_visit_column(column, self),
```

Comment:
Do `EmptyArray` and `EmptyMap` belong to `SimpleType` ?

Reply:
Doesn't belong, I think they can be abstracted as ZeroSizeType trait